juju/juju
# refactor(facade): use model config service in provisioner facade

This PR refactors the provisioner facade and the common ToolsGetter to use model services. The introduced services are model info/config/agent as appropriate.

Additional changes:
- Remove the unused configGetter from common.ToolsFinder.
- Rename `stdcontext` to `context` in some places.
- Be safer when casting the value in `Config.SSLHostnameVerification` to avoid panics.
- Saving machine data to dqlite should be idempotent.

## Checklist




## QA steps

Affected facades and methods are:
- Provisioner: GetContainerProfileInfo, ModelUUID, ProvisioningInfo
- Upgrader: Tools

To test the provisioner, bootstrap Juju on AWS:
```
$ juju bootstrap aws aws
$ juju add-model m
$ juju add-machine
created machine 0
$ juju deploy ubuntu --to lxd:0
```
Check that ubuntu comes up correctly and doesn't crash the container or the host machine.

Set the resource tags in model config:
```
juju model-config resource-tags="foo=bar"
```

Create a storage pool:
```
juju create-storage-pool ebs ebs size=1G
```

Provision a new machine with an attached volume:
```
juju add-machine --disks="ebs,100M,1"
```
and check on the AWS console that the provisioned machine and attached volume both have the specified resource tags.

**TODO:** test upgrader worker

## Links

**Jira card:** [JUJU-6056](https://warthogs.atlassian.net/browse/JUJU-6056)

[JUJU-6056]: https://warthogs.atlassian.net/browse/JUJU-6056?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
